pennylane/templates/subroutines/trotter.py

```diff
@@ -236,6 +240,30 @@ def queue(self, context=qml.QueuingManager):
         context.append(self)
         return self
 
+    def resources(self) -> Resources:
+        """The resource requirements for a given instance of the Suzuki-Trotter product.
+
+        Returns:
+            Resources: The resources for an instance of TrotterProduct.
+        """
+        with qml.QueuingManager.stop_recording():
+            decomp = self.compute_decomposition(*self.parameters, **self.hyperparameters)
+
+        num_wires = len(self.wires)
+        num_gates = len(decomp)
+
+        unique_operation_decomp = (copy.deepcopy(op) for op in decomp)
```

Comment:
I did ask it before, but once again, why do we call this `unique_operation`? It simply makes a generator. 🤔 

![image](https://github.com/PennyLaneAI/pennylane/assets/23234279/516e9390-f691-4229-8f3e-589b72b3d324)

Reply:
What I mean by the name `unique_operation_decomp` is that this is a decomposition where each operation has been deep_copied so each one is unique (seperate hash) as opposed to the previous decomp where we were using multiple copies of the same operator.  

We need "unique" deep-copies in order to properly count resources. Due to some legacy code in the circuit graph module.

Reply:
I don't think that the deep copying changes the hash - 
![image](https://github.com/PennyLaneAI/pennylane/assets/23234279/af17ceab-26b0-44eb-b0fb-dfc3d4824663).
Anyway, since it is legacy code, it doesn't matter much!